tvb_storage/tvb/storage/storage_interface.py

```diff
@@ -388,24 +373,107 @@ def build_data_export_folder(self, data, export_folder):
+        self.write_zip_folder(result_path, project_folder, linked_paths, op, folders_to_exclude)
 
         # Make sure the Project.xml file gets copied:
         self.logger.debug("Done, closing")
 
         return result_path
+
+    def export_simulator_configuration(self, burst, all_view_model_paths, all_datatype_paths, zip_filename):
+        """
+        This method is used to export a simulator configuration as a ZIP file
+        :param burst: BurstConfiguration of the simulation to be exported
+        :param all_view_model_paths: a list of paths to all view model files of the simulation
+        :param all_datatype_paths: a list of paths to all datatype files of the simulation
+        :param zip_filename: name of the file to be exported
+        """
+
+        tmp_export_folder = self.__build_data_export_folder(burst, self.EXPORT_FOLDER)
+        tmp_sim_folder = os.path.join(tmp_export_folder, self.EXPORTED_SIMULATION_NAME)
+
+        if not os.path.exists(tmp_sim_folder):
+            os.makedirs(tmp_sim_folder)
+
+        for vm_path in all_view_model_paths:
+            dest = os.path.join(tmp_sim_folder, os.path.basename(vm_path))
+            self.copy_file(vm_path, dest)
+
+        for dt_path in all_datatype_paths:
+            dest = os.path.join(tmp_sim_folder, self.EXPORTED_SIMULATION_DTS_DIR, os.path.basename(dt_path))
+            self.copy_file(dt_path, dest)
+
+        result_path = os.path.join(tmp_export_folder, zip_filename)
+        self.write_zip_folder(result_path, tmp_sim_folder)
+        self.remove_folder(tmp_sim_folder)
+
+        return result_path
+
+    def export_datatypes(self, dt_path_list, data, download_file_name):
+        """
+        This method is used to export a list of datatypes as a ZIP file.
+        :param dt_path_list: a list of paths to be exported (there are more than one when exporting with links)
+        :param data: data to be exported
+        :param download_file_name: name of the zip file to be downloaded
+        """
+
+        export_folder = self.__build_data_export_folder(data, self.EXPORT_FOLDER)
+        file_destination = None
+
+        for dt_path in dt_path_list:
+            file_destination = os.path.join(export_folder, os.path.basename(dt_path))
+            if not os.path.exists(file_destination):
+                self.copy_file(dt_path, file_destination)
+            self.get_storage_manager(file_destination).remove_metadata('parent_burst', check_existence=True)
+
+        if len(dt_path_list) == 1:
+            return file_destination
+
+        export_data_zip_path = os.path.join(os.path.dirname(export_folder), download_file_name)
+        self.write_zip_folder(export_data_zip_path, export_folder)
+        return export_data_zip_path
+
+    def export_datatypes_structure(self, all_datatypes, data, download_file_name, project_name):
+        """
+        This method is used to export a list of datatypes as a ZIP file. It is only used during normal tvb exporting.
```

Comment:
This method is used to export a list of datatypes as a ZIP file, while preserving the folder structure (eg: operation folders). It is only used during normal tvb exporting for datatype groups.